src/core/include/mp-units/bits/sudo_cast.h

```diff
@@ -79,10 +74,53 @@ struct conversion_value_traits {
   static constexpr Magnitude auto num = _numerator(M);
   static constexpr Magnitude auto den = _denominator(M);
   static constexpr Magnitude auto irr = M * (den / num);
+  static constexpr auto ratio = [] {
+    if constexpr (std::is_integral_v<T>) {
+      using U = long double;
+      return detail::fixed_point<T>{_get_value<U>(num) / _get_value<U>(den) * _get_value<U>(irr)};
+    } else {
+      return _get_value<T>(num) / _get_value<T>(den) * _get_value<T>(irr);
+    }
+  }();
+  static constexpr bool value_increases = ratio >= T{1};
+
+  template<typename V>
+  static constexpr auto scale(V value)
+  {
+    if constexpr (std::is_integral_v<T>) {
```

Comment:
```suggestion
    if constexpr (requires { ratio.scale(value); }) {
```

or

```suggestion
    if constexpr (is_specialization_of<T, fixed_point>) {
```

Reply:
Nope. I don't think either of these suggestions are good choices here, but we may indeed want to refactor here. Right now, this `std::is_integral_v<T>` check really represents the `treat_as_floating_point` behavioural choice here. So this `if` here should really be replaced by a customisation point for custom representation types. I would decidedly *not* try to "guess" a scaling method here, but rather make it an explicit choice somewhere in the scaling machinery.

Instead, I believe I should refactor the `conversion_value_traits` a bit: In fact, these are really `scaling_traits` in the `Scalable` concept discussed recently. Right now, `conversion_value_traits` are separated by the properties of the scaling magnitude, but I believe this should be reversed: The "outer level" is the potentially customisable scaling behaviour of the representation types. The implementation of that scaling behaviour for different magnitude types is almost an implementation detail and thus should be at the "inner level".